db/view/view.cc

```diff
@@ -1793,6 +1793,8 @@ future<> view_update_generator::mutate_MV(
                 }
                 tracing::trace(tr_state, "Successfully applied local view update for {}", my_address);
                 return make_ready_future<>();
+            }).finally([this] {
+                _proxy.local().update_view_update_backlog();
```

Comment:
Why not do it in the `then_wrapped` above? With the additional `finally` you are forced to do an additional allocation.

Reply:
The update needs to be executed after the `sem_units` are destroyed which is a bit problematic because they're shared in both local and remote update path. I found we can do it a bit more verbosely without the additional allocation in the rebase.